src/Data/Conduit/Tar/Unix.hs

```diff
@@ -91,6 +93,26 @@ restoreFileInternal lenient fi@FileInfo {..} = do
 #endif
                 return $ fst $ partitionEithers [eExc1, eExc2]
             unless (null excs) $ yield (return (fi, excs))
+        FTHardLink link -> do
+            excs <- liftIO $ do
+                let linkedFp = decodeFilePath link
+                linkedFileExists <- Posix.fileExist linkedFp
+                -- If the linked file does not exist (yet), we cannot create a hard link.
+                -- Try to "pre-create" it.
+                when (not linkedFileExists && lenient) $ do
```

Comment:
Can we push the fileExist test under the when lenient - that gives a performance boost for non-lienient things.

Reply:
Done.